src/components/accessibility/partitions_data_table.tsx

```diff
@@ -0,0 +1,145 @@
+import {
+  A11ySettings,
+  DEFAULT_A11Y_SETTINGS,
+  getA11ySettingsSelector,
+} from '../../state/selectors/get_accessibility_config';
+import { getInternalIsInitializedSelector, InitStatus } from '../../state/selectors/get_internal_is_intialized';
+import { getSettingsSpecSelector } from '../../state/selectors/get_settings_specs';
+import { isNil } from '../../utils/common';
+
+interface ScreenReaderPartitionTableProps {
+  a11ySettings: A11ySettings;
+  partitionData: PartitionData;
+  debug: SettingsSpec['debug'];
+}
+
+// this currently limit the number of pages shown to the user
+const TABLE_PAGINATION = 20;
+
+const ScreenReaderPartitionTableComponent = ({
+  a11ySettings,
+  partitionData,
+  debug,
+}: ScreenReaderPartitionTableProps) => {
+  const [count, setCount] = useState(1);
+  const tableRowRef = createRef<HTMLTableRowElement>();
+  const { tableCaption } = a11ySettings;
+
+  const rowLimit = TABLE_PAGINATION * count;
+  const handleMoreData = () => {
+    setCount(count + 1);
+    // generate the next group of data
+    if (tableRowRef.current) {
+      tableRowRef.current.focus();
+    }
+  };
+
+  const { isSmallMultiple, data, hasMultipleLayers } = partitionData;
+  const tableLength = data.length;
+  const showMoreRows = rowLimit < tableLength;
+
+  return (
+    <div className={`echScreenReaderOnly ${debug ? 'echScreenReaderOnlyDebug' : ''} echScreenReaderTable`}>
+      <table>
+        <caption>
+          {isNil(tableCaption)
+            ? `The table ${
+                showMoreRows
+                  ? `represents only ${rowLimit} of the ${tableLength} data points`
+                  : `fully represents the dataset of ${tableLength} data point${tableLength > 1 ? 's' : ''}`
+              }`
+            : tableCaption}
+        </caption>
+        <thead>
+          <tr>
+            {isSmallMultiple && <th scope="col">Small multiple title</th>}
+            {hasMultipleLayers && <th scope="col">Depth</th>}
+            <th scope="col">Label</th>
+            {hasMultipleLayers && <th scope="col">Parent</th>}
+            <th scope="col">Value</th>
+            <th scope="col">Percentage</th>
+          </tr>
+        </thead>
+
+        <tbody>
+          {partitionData.data
+            .slice(0, rowLimit)
+            .map(({ panelTitle, depth, label, parentName, valueText, percentage }, index) => {
+              return (
+                <tr key={`row--${index}`} ref={rowLimit === index ? tableRowRef : undefined} tabIndex={-1}>
+                  {isSmallMultiple && <td>{panelTitle}</td>}
+                  {hasMultipleLayers && <td>{depth}</td>}
+                  <th>{label}</th>
+                  {hasMultipleLayers && <td>{parentName}</td>}
+                  <td>{valueText}</td>
+                  <td>{percentage}</td>
+                </tr>
+              );
+            })}
+        </tbody>
+        {showMoreRows && (
+          <tfoot>
+            <tr>
+              <td>
```

Comment:
You'll need to calculate how many columns will get rendered and to grow the table cell with the button.

Looks like it's 3 (label, value, and percentage) plus 1 if there are multiple layers, plus 1 if there are small multiples, and plus one if there are multiple layers.

```suggestion
              <td colspan={totalCols}>
```

Reply:
:mindblown: Thank you!!! 82e8219 for changes 🙇🏻‍♀️